kuma/api/v1/views.py

```diff
@@ -365,13 +368,20 @@ def send_subscriptions_feedback(request):
     return HttpResponse(status=204)
 
 
-@api_view(["POST"])
-def create_subscription(request):
+@api_view(["POST", "GET"])
```

Comment:
any reason to have one fn for both? Imo they don't share enough functionality that I would've opted for that.

Reply:
How else are you supposed to do it? There is one option, but I'm not convinced it's better. 
One option would be to use class-based views so that can have one method called `post` and another method called `get`. 
But we don't use class-based views in Kuma. ...except in some areas where we integrate with django_restframework. 

What would be "ideal" would be if Django allowed you to specify this in `urls.py`. Like this:

```python
urlpatterns = [
    path("subscriptions/", methods.GET, views.get_subscriptions),
    path("subscriptions/", methods.POST, views.create_subscription),
]
```
but that's not how Django works. 

It's never going to get sane because you can change two variables at the same time: the URL and the method. Sometimes, you want the same URL and a list of methods. Sometimes the other way around. 

What would you have opted for?

Reply:
By the way, I think I can see why Django works like this. Back in the day, when there wasn't any SPAs and where Django got to do everything and almost always with Django forms. You'd see this pattern a lot:

```
def some_view_that_shows_and_creates(request):
    if request.method == "POST":
        form = FeedbackForm(request.POST)
        if form.is_valid():
            form.save()
            # redirect to prevent browser refreshes submitting
           return redirect(url() + "?created=true")
    else:
         form = FeedbackForm()
    return render(request, 'template.html', {'form': form})
```
this way, the `form`, when used inside `template.html` it's sometimes bound by data and validation and sometimes unbound.

Reply:
Ah dang, didn't consider Django being in the way of doing it right, sorry 🙈 

django-rest-framework has a feature for handling it, but that seems to necessitate using classes, which I find even less exciting:
https://www.django-rest-framework.org/api-guide/viewsets/

So yeah, sorry, not as actionable I'd have hoped this suggestion would be.

Reply:
Oh right, I remember that pattern. It does make sense for that. Though I think a router (aka urlpatterns) where you'd also declare request method would make more sense (which is how Rails does it). Then you can have both the classic server rendered and the SPA cake 🤷‍♂️ But I guess we have to work with the Django we have, not the one I want 😁

Reply:
I'm pretty confident that the REST'y use of 
```
GET  /api/v1/subscriptions/
POST /api/v1/subscriptions/
```
is sane. I don't think Kuma (or API v1) is ever trying to be accurate about how it does REST but it's a pretty common pattern to use the same URL but different methods.
...instead of doing something like:
```
GET  /api/v1/subscriptions/list/
POST /api/v1/subscriptions/create/
```

Reply:
oh yeah no, I'm in complete agreement with what the URLs look like, it's just the view unification where I think django does the wrong thing.